tests/test_model.py

```diff
@@ -2435,3 +2437,66 @@ def test_resolve(self):
         assert complaint.resolved != None
         assert abs(datetime.datetime.utcnow() - complaint.resolved).seconds < 3
 
+
+class TestCollection(DatabaseTest):
+
+    def test_encrypts_client_secret(self):
+        collection, new = get_one_or_create(
+            self._db, Collection, name=u"Test Collection", client_id=u"test",
+            client_secret=u"megatest"
+        )
+        assert collection.client_secret != u"megatest"
+        eq_(True, collection.client_secret.startswith("$2a$"))
+
+    def test_register(self):
+        collection, plaintext_secret = Collection.register(
+            self._db, u"A Library"
+        )
+
+        # It creates a collection with client_details.
+        assert collection.client_id and collection.client_secret
+
+        # It returns nothing if the name is already taken.
+        assert_raises(ValueError, Collection.register, self._db, u"A Library")
+
+        # It creates a DataSource for the Collection by default.
+        assert get_one(self._db, DataSource, name=collection.name)
+
+        # It can also create a Collection without a Data Source.
+        collection, plaintext_secret = Collection.register(
+            self._db, u"A Non-Library", is_source=False
+        )
+        source = get_one(self._db, DataSource, name=collection.name)
+        eq_(None, source)
+
+    def test_authenticate(self):
+        collection = self._collection()
+
+        result = Collection.authenticate(self._db, u"abc", u"def")
+        eq_(collection, result)
+
+        result = Collection.authenticate(self._db, u"abc", u"bad_secret")
+        eq_(None, result)
+
```

Comment:
Could we add 
result = Collection.authenticate(self._db, u"bad_client_id", u"def")
?